src/usdb_syncer/gui/song_table/table_model.py

```diff
@@ -234,3 +234,7 @@ def yes_no_str(yes: bool) -> str:
 @cache
 def optional_check_icon(yes: bool) -> QIcon | None:
     return QIcon(":/icons/tick.png") if yes else None
+
+
```

Comment:
```suggestion

@cache
```
Icon generation seems to be relatively expensive, so we should cache it.